src/dune_engine/format_config.ml

```diff
@@ -84,43 +84,72 @@ module Enabled_for = struct
     | _ -> false
 end
 
-type 'enabled_for generic_t =
+type ('enabled_for, 'files) very_generic_t =
   { loc : Loc.t
   ; enabled_for : 'enabled_for
+  ; files : 'files
   }
 
-type t = Enabled_for.t generic_t
+type 'files generic_t = (Enabled_for.t, 'files) very_generic_t
+
+type t = unit generic_t
 
 let includes t lang = Enabled_for.includes t.enabled_for lang
 
-let to_dyn { enabled_for; loc = _ } =
+let to_dyn { enabled_for; loc = _; files = _ } =
   let open Dyn in
   record [ ("enabled_for", Enabled_for.to_dyn enabled_for) ]
 
 let dparse_args =
   let+ loc = loc
   and+ enabled_for = fields Enabled_for.field_ext in
-  ({ loc; enabled_for }, [])
+  ({ loc; enabled_for; files = () }, [])
 
-let dune2_record_syntax =
-  let+ ef = Enabled_for.field in
-  match ef with
-  | Some l -> Enabled_for.Only (Language.Set.of_list l)
-  | None -> All
+let enabled_for_all =
+  { loc = Loc.none; enabled_for = Enabled_for.All; files = () }
 
-let dune2_dec =
-  let+ loc = loc
-  and+ enabled_for =
-    keyword "disabled"
-    >>> return (Enabled_for.Only Language.Set.empty)
-    <|> fields dune2_record_syntax
-  in
-  { loc; enabled_for }
+let disabled =
+  { loc = Loc.none
+  ; enabled_for = Enabled_for.Only Language.Set.empty
+  ; files = ()
+  }
 
-let enabled_for_all = { loc = Loc.none; enabled_for = Enabled_for.All }
+module Generic = struct
+  type 'files t = 'files generic_t
+
+  let dune2_record_syntax ~files =
+    let+ files = files
+    and+ ef = Enabled_for.field in
+    ( files
+    , match ef with
+      | Some l -> Enabled_for.Only (Language.Set.of_list l)
+      | None -> All )
+
+  let dune2_dec ~default_files ~files =
+    let+ loc = loc
+    and+ files, enabled_for =
+      peek_exn >>= function
```

Comment:
I think that this `peek_exn` can be moved to top, that way you don't have to deal with tuples